dev-utils/src/commands/init_command.rs

```diff
@@ -0,0 +1,225 @@
+            "{}",
+            serde_json::to_string_pretty(&self.genesis_dev).unwrap()
+        );
+        File::create(&path)
+            .map(|mut file| {
+                let note = serde_json::to_string(&self.genesis_dev).unwrap();
+                file.write_all(note.as_bytes()).unwrap();
+            })
+            .unwrap();
+
+        // app_hash,  app_state
+        let result = GenesisCommand::generate(&path.to_path_buf()).unwrap();
+        println!("genesis_time( {} )=", self.genesis_time);
+
+        self.app_hash = json!(result.0);
+        self.app_state = serde_json::from_str(&result.1).unwrap();
+    }
+    pub fn get_tendermint_filename(&self) -> String {
+        format!(
+            "{}/.tendermint/config/genesis.json",
+            dirs::home_dir().unwrap().to_str().unwrap()
+        )
+        .to_string()
+    }
+    pub fn read_tendermint_genesis(&mut self) {
+        // check whether file exists
+        let _dummy = fs::read_to_string(&self.get_tendermint_filename()).and_then(|contents| {
+            println!("current tendermint genesis={}", contents);
+            let json: serde_json::Value = serde_json::from_str(&contents).unwrap();
+            let pub_key = &json["validators"][0]["pub_key"]["value"];
+            self.tendermint_pubkey = pub_key.as_str().unwrap().to_string();
+            Ok(())
+        });
+    }
+    pub fn write_tendermint_genesis(&self) {
+        println!("write genesis to {}", self.get_tendermint_filename());
+
+        let app_hash = self.app_hash.clone();
+        let app_state = self.app_state.clone();
+        let gt = self.genesis_time.clone();
+
+        let mut json_string = String::from("");
+        let _dummy = fs::read_to_string(&self.get_tendermint_filename())
+            .and_then(|contents| {
+                let mut json: serde_json::Value = serde_json::from_str(&contents).unwrap();
+                let obj = json.as_object_mut().unwrap();
+                obj["app_hash"] = app_hash;
+                obj.insert("app_state".to_string(), json!(""));
+                obj["app_state"] = app_state;
+                obj["genesis_time"] = gt;
+                json_string = serde_json::to_string_pretty(&json).unwrap();
+                println!("{}", json_string);
+
+                File::create(&self.get_tendermint_filename())
+            })
+            .map(|mut file| file.write_all(json_string.as_bytes()))
+            .map(|_e| {
+                println!(
+                    "writing tendermint genesis OK {}",
+                    self.get_tendermint_filename()
+                );
+            })
+            .map_err(|e| {
+                println!("Error={}", e);
+            });
+    }
+    pub fn prepare_tendermint(&self) {
+        // check whether file exists
+        let _dummy = fs::read_to_string(&self.get_tendermint_filename()).map_err(|_e| {
+            // file not exist
+            Command::new("tendermint").args(&["init"]).output().unwrap();
```

Comment:
if the person executing this doesn't have `tendermint` binary in PATH, it'll end up with a quite confusing error:
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', src/libcore/result.rs:999:5
```

I think it'll be good to have either proper error handling or (simpler), just use "expect" instead of "unwrap" for some meaningful panic messages

Reply:
agree, i will add error handler

Reply:
fixed